### PR TITLE
Jarrel/use dockertest for test

### PR DESCRIPTION
**Pull request by jarrel-b**
**Changes**
Unit tests start a new container per test file (the tests were super flaky if containers were started each test), instead of using the local containers and volumes for testing. The `setupDoubles` fixture is used to set up the containers and tear them down and patches the environment. The `setupTest` fixture sets up the server and wipes the tables after each unit test. 

Also moved some functions previously in `server` to `docker` and `db` so they can be re-used.

**Testing**
Ran `go test -short ./...`

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	for _, cmd := range res {
+		if cmd.Cmd == "FROM" {
+			return getImageAndVersion(cmd.Value[0])
+		}
+	}
+
+	return nil, errors.New("no `FROM` directive found in dockerfile")
+}
+
+func InitPostgres(composePath string) *dockertest.Resource {
```
Comment (radazen) [inline docker/docker.go:118]: How would you feel about getting rid of the `composePath` parameter and just defaulting to the `docker-compose.yml` file at the project root? Or splitting it into two functions so we have an `InitPostgres` function that takes no parameters and calls `InitPostgresWithComposePath` or something with the default docker-compose file location? I like having the path be configurable, but it seems like all uses of this function pass the same `../../../docker-compose.yml` style parameter, and each of them is responsible for figuring out the correct level of `../..` based on how deeply the particular test file is nested.

**Review by radazen — APPROVED** [review]
I love everything about this! 🎉🎉🎉 Left one small comment re: docker-compose.yml path.

🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌

**Review by Robinnnnn — APPROVED** [review]

**Review by Robinnnnn — APPROVED** [review]

File: docker/docker.go (hunk 1 above)
Comment (radazen) [reply docker/docker.go:118]: Ah, that makes sense. I tried some googling to see if there's an easy way to get the project root, but it doesn't seem like there's a reliable way to do it. Maybe check for `./docker-compose.yml`, `../docker-compose.yml`, `../../docker-compose.yml`, etc until we find a file or hit some depth threshold? I'm mostly imagining myself trying to figure out why tests are broken some day when I refactor stuff and move files around.

Comment (jarrel-b) [reply docker/docker.go]: ooo like this idea, added this for finding `docker-compose.yml` and also for finding the migration files to seed the containers.

**Review by radazen — APPROVED** [review]
Yup, this is awesome.

**Pull request merged**